exercises/practice/bob/.approaches/introduction.md

```diff
@@ -0,0 +1,115 @@
+# Introduction
+
+There are various idiomatic approaches to solve Bob.
+A basic approach can use a series of `if` statements to test the conditions.
+Or a [switch][switch] on a [tuple][tuple] of the conditions can be used.
+An array can contain answers from which the right response is selected by an index calculated from scores given to the conditions.
+
+## General guidance
+
+Regardless of the approach used, some things to look for include
+
+- If the input is trimmed, [Trim][trim] only once.
+
+- Use the [EndsWith][endswith] `String` method instead of checking the last character by index for `?`.
+
+- Don't copy/paste the logic for determining a shout and for determing a question into determing a shouted question.
+Combine the two determinations instead of copying them.
+Not duplicating the code will keep the code [DRY][dry].
+
+- Perhaps consider making `IsQuestion` and `IsShout` values set once instead of functions that are possibly called twice.
+
+- If an `if` statement can return, then an `else if` or `else` is not needed.
+Execution will either return or will continue to the next statement anyway.
+
+- If the body of an `if` statement is only one line, curly braces aren't needed.
+Some teams may still require them in their style guidelines, though.
+
+## Approach: `if` statements
+
+```csharp
+using System.Linq;
+public static class Bob
+{
+    public static string Response(string message)
+    {
+        if (IsSilence(message))
+            return "Fine. Be that way!";
+
+        if (IsYell(message) && IsQuestion(message))
+            return "Calm down, I know what I'm doing!";
+
+        if (IsYell(message))
+            return "Whoa, chill out!";
+
+        if (IsQuestion(message))
+            return "Sure.";
+
+        return "Whatever.";
+    }
+
+    private bool IsSilence(string message) {
```

Comment:
I'm gonna be annoying, but putting the opening curly brace on the same line as the method definition is definitely not a C# style that people use.